docs/README-develop-best-practices.md

```diff
@@ -43,132 +44,124 @@ This document is intended to assist in developing cisco_node_utils API's that ar
 
 ## YAML Best Practices:
 
-### <a name="yaml1">Y1: All yaml feature entries should be kept in alphabetical order.
+### <a name="yaml1">Y1: One feature per YAML file
 
-Please keep all feature names in alphabetical order, and all options under a feature in alphabetical order as well. As YAML permits duplicate entries (in which case the last entry overrides any earlier entries), keeping a consistent order helps to prevent accidentally introducing such duplication.
+Each YAML file should define a single 'feature' (a closely related set of configuration properties). Don't create "one YAML file to rule them all".
 
-Top level features in alpabetical order:
+### <a name="yaml2">Y2: All attribute entries must be kept in alphabetical order.
 
-```
-aaa_authentication_login:
-...
-dnsclient:
-...
-interface:
-```
+All attribute entries in a given YAML file must be kept in alphabetical order. As YAML permits duplicate entries (in which case the last entry overrides any earlier entries), keeping a consistent order helps to prevent accidentally introducing such duplication.
 
-Options under a feature:
-
-```
-interface:
-  access_vlan:
-    ...
-  all_interfaces:
-    ...
-  create:
-    ...
-  description:
-    ...  
-```
+This rule is enforced by the `Cisco::CommandReference` class itself - it will raise an exception if it detects any out-of-order entries.
 
-### <a name="yaml2">Y2: Use *regexp* anchors where needed for `config_get` and `config_get_token` entries.
+### <a name="yaml3">Y3: Use *regexp* anchors where needed for `config_get` and `config_get_token` entries.
 
 Please use *regexp* anchors `^$` to ensure you match the correct feature information in the `show` output.
 
-```
-syslog_settings:
-  timestamp:
-    config_get: "show running-config all | include '^logging timestamp'"
-    config_get_token: '/^logging timestamp (.*)$/'
-    config_set: '<state> logging timestamp <units>'
-    default_value: 'seconds'
+```yaml
+# syslog_settings.yaml
+timestamp:
+  config_get: "show running-config all | include '^logging timestamp'"
+  config_get_token: '/^logging timestamp (.*)$/'
+  config_set: '<state> logging timestamp <units>'
+  default_value: 'seconds'
 ```
 
-### <a name="yaml3">Y3: Avoid nested optional matches.
+### <a name="yaml4">Y4: Avoid nested optional matches.
```

Comment:
Since you are already modifying this file, could you add an example for this?  I meant to, but missed it.